spring-cloud-contract-wiremock/src/main/java/org/springframework/cloud/contract/wiremock/file/ResourcesFileSource.java

```diff
@@ -34,28 +34,37 @@
 
 /**
  * @author Dave Syer
- *
+ * @author Pei-Tang Huang
  */
 public class ResourcesFileSource implements FileSource {
 
 	private final FileSource[] sources;
 
-	public ResourcesFileSource(Resource[] resources) {
-		this.sources = new FileSource[resources.length];
+	public ResourcesFileSource(Resource... resources) {
+		this(toSources(resources));
+	}
+
+	private ResourcesFileSource(FileSource... sources) {
+		this.sources = sources;
+	}
+
+	private static FileSource[] toSources(Resource[] resources) {
+		FileSource[] sources = new FileSource[resources.length];
 		for (int i = 0; i < resources.length; i++) {
 			Resource resource = resources[i];
 			if (resource instanceof ClassPathResource) {
 				ClassPathResource classes = (ClassPathResource) resource;
-				this.sources[i] = new ClasspathFileSource(classes.getPath());
+				sources[i] = new ClasspathFileSource(classes.getPath());
```

Comment:
Could you please leave `this.` ?

Reply:
This method was extracted from the constructor as a static helper method converting resrouce[] to fileSource[], and have no access to the instance field.
Should I rename it to something else, or inline it back to the origin method?

Reply:
you can call `ResourcesFileSource.this` or pass the `sources[i]` to a `final` variable. We have a checkstyle rule that checks this - that's why I'm asking for this change.

Reply:
@marcingrzejszczak , Sorry, I ran checkstyle checks using https://raw.githubusercontent.com/spring-cloud/spring-cloud-build/v1.3.5.RELEASE/spring-cloud-build-tools/src/main/resources/checkstyle.xml , but my CheckStyle-IDEA plugin with CheckStyle 8.2 shows `Checkstyle found no problems in the file(s)` . Am I missing something?

Reply:
Interesting... You can see there that there is this `requireThis` entry. I don't know why that doesn't work. Can you try from Maven?

Reply:
Result of my `mvn checkstyle:check` run:
![image](https://user-images.githubusercontent.com/638068/30600875-244b7434-9d93-11e7-96ae-ff8e421cf9f5.png)

One more thing, spring-cloud-contract-wiremock's `name` in POM.xml is in a different naming scheme from other modules.